### PR TITLE
[#42]refactor: Update firmwareSimulator and proto files

### PR DESCRIPTION
**Pull request by Mythois**
update the firmwareSimulator to handle habits of type timer. Also, update proto, desc and pb2.py files to match new protocol buffer message format

## Related issues
Issue #

## Checklist

- [x] I have added comments to code where needed
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have tested that there are no new bugs
- [ ] I have added tests that prove my fix is effective or that my feature works (Optional before testing is discovered)

### CODE REVIEW

File: firmwareSimulator/habitTracker.py [hunk 1]
```diff
@@ -127,37 +127,74 @@ def on_lifecycle_connection_failure(self, lifecycle_connection_failure: mqtt5.Li
...
+                            topic = mqtt_topic,
+                            payload = payload,
+                            qos = mqtt5.QoS.AT_LEAST_ONCE
+                        ))
+                        print("payload: {}".format(payload))
+                        print("Sent to topic {}".format(mqtt_topic))
+                    case _:
+                        raise Exception("The provided format must either be JSON or proto_buff")
+            
+            case "timer":
```
Comment (EskildVeren) [inline firmwareSimulator/habitTracker.py:165]: rename "timer" to time

File: firmwareSimulator/habitTracker.py [hunk 2]
```diff
@@ -127,37 +127,74 @@ def on_lifecycle_connection_failure(self, lifecycle_connection_failure: mqtt5.Li
...
+                message = {
+                    "device_timestamp": int(round(datetime.timestamp(datetime.now()))),
+                    "habit_id": habit_id,
+                    "start_timestamp": int(round(datetime.timestamp(datetime.now()))),
```
Comment (EskildVeren) [inline firmwareSimulator/habitTracker.py:169]: Make changes so start isnt after stop

Comment (Mythois) [reply firmwareSimulator/habitTracker.py]: Should be fixed now. 
But best to have a new look at my code XD

File: firmwareSimulator/habitTracker.py (hunk 1 above)
Comment (Mythois) [reply firmwareSimulator/habitTracker.py]: Should be fixed now. 
But best to have a new look at my code XD

**Comment by Mythois** [comment]
I have rewritten the publish_message method to make it cleaner. While doing so I have incorporated your feedback. 
Minor changes have been done to other parts of the code in order for the rewrite of the publish_message method to work as intended.

I have also checked that the refactored method still works with AWS MQTT as before.

**Review by EskildVeren — APPROVED** [review]

**Pull request merged**